### PR TITLE
fix: ensure children containers are killed for container sets. Fixes #10491

### LINKED ISSUE #10491: Workflow hangs after clicking stop and terminate
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

When clicking stop or terminate within the UI, we are noticing that workflows are left hanging in a running state. The only way to get them to actually stop is to delete the workflow. This causes a problem for users who want to stop the workflow but keep the record in their archives

Here's a list of some other issues that I think might be related. I will keep this list up to date as more arise:
- https://github.com/argoproj/argo-workflows/issues/10063
- https://github.com/argoproj/argo-workflows/issues/8680
- https://github.com/argoproj/argo-workflows/issues/9934

### Version

latest

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
Unfortunately, this happens intermittently. I will update this issue with some videos and see if I can find some workflows that consistently trigger it using only public images
```

### Logs from the workflow controller

```text
see above
```

### Logs from in your workflow's wait container

```text
See above
```

### PR DESCRIPTION
**Pull request by isubasinghe**
Fixes #10491

### CODE REVIEW

File: workflow/controller/exec_control.go [hunk 1]
```diff
@@ -82,10 +82,23 @@ func (woc *wfOperationCtx) handleExecutionControlError(nodeID string, wfNodesLoc
 	node := woc.wf.Status.Nodes[nodeID]
 	woc.markNodePhase(node.Name, wfv1.NodeFailed, errorMsg)
 
+	children := []wfv1.NodeStatus{}
```
Comment (alexec) [inline workflow/controller/exec_control.go:85]: This looks like a useful utility function,:

- Is there already a similar function?
- If not, extract it, e.g. `woc.wf.Status.Nodes.GetTransientChildren(nodeID)`

Extracting it to a function will:

- Allow reuse.
- Write test.

**Review by alexec — CHANGES_REQUESTED** [review]
Please dismis this review to request re-review.

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 2]
```diff
@@ -1735,6 +1735,30 @@ func (s Nodes) Children(parentNodeId string) Nodes {
 	return childNodes
 }
 
+func (s Nodes) NestedChildrenStatus(parentNodeId string) ([]NodeStatus, error) {
```
Comment (alexec) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:1738]: now, please add tests!

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 3]
```diff
@@ -1735,6 +1735,30 @@ func (s Nodes) Children(parentNodeId string) Nodes {
...
+	q := []NodeStatus{parentNode}
+
+	for {
+		if len(q) <= 0 {
+			break
+		}
+		childNode := q[0]
+		q = q[1:]
+		for _, nodeID := range childNode.Children {
+			q = append(q, s[nodeID])
```
Comment (alexec) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:1754]: do we need to go down the tree?

**Review by alexec — APPROVED** [review]

**Review by sarabala1979 — APPROVED** [review]

**Pull request merged**